metis/v1/identity/costobjects/results.go

```diff
@@ -0,0 +1,67 @@
+// Copyright 2023 SAP SE
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package costobjects
+
+import (
+	"github.com/gophercloud/gophercloud"
+	"github.com/gophercloud/gophercloud/pagination"
+
+	v1 "github.com/sapcc/gophercloud-sapcc/metis/v1"
+)
+
+type CostObject struct {
+	Name string `json:"name"`
+	Type string `json:"type"`
+}
+
+// Extract accepts a Page struct, specifically an v1.CommonPage struct,
+// and extracts the elements into a slice of CostObjects structs.
+func Extract(r pagination.Page) ([]CostObject, error) {
+	var s struct {
+		CostObjects []CostObject `json:"items"`
+	}
+	if err := (r.(v1.CommonPage)).ExtractInto(&s); err != nil {
+		return []CostObject{}, err
```

Comment:
is there a reason to return non-nil empty object?

Reply:
No reason, have changed it for all Extract methods.